tsdb/meta.go

```diff
@@ -796,9 +798,12 @@ func (m *Measurement) TagSets(shardID uint64, dimensions []string, condition inf
 		tags := make(map[string]string, len(dimensions))
 
 		// Build the TagSet for this series.
+		s.mu.RLock()
 		for _, dim := range dimensions {
-			tags[dim] = s.Tags.GetString(dim)
+			tags[dim] = s.tags.GetString(dim)
```

Comment:
I was trying to avoid extra locking but I can change it to `s.GetTagString()`. I'd rather not generate a bunch of garbage by calling `Tags()`.